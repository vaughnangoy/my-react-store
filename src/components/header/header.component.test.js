import React from 'react';
import { cleanup, render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import Header from './header.component';

import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { Router } from 'react-router-dom';
import { createBrowserHistory } from 'history';

const generateComponent = props => {
    const history = createBrowserHistory();
    const store = createStore(() => {}, {});
    return (
        <Provider store={store}>
            <Router history={history}>
                <>
                    <Header {...props} />
                </>
            </Router>
        </Provider>
    );
};

describe('Cart Dropdown', () => {
    afterEach(cleanup);

    it('should render', () => {
        const { asFragment } = render(generateComponent());
        expect(asFragment()).toMatchSnapshot();
    });

    it('should render the correct links', () => {
        const { getByText } = render(generateComponent());
        expect(getByText('SHOP')).toBeInTheDocument();
        expect(getByText('CONTACT')).toBeInTheDocument();
        expect(getByText('LOGIN')).toBeInTheDocument();
    });

    it('should render the correct links when logged out', () => {
        const { getByText } = render(generateComponent({ currentUser: {} }));
        expect(getByText('SHOP')).toBeInTheDocument();
        expect(getByText('CONTACT')).toBeInTheDocument();
        expect(getByText('LOGOUT')).toBeInTheDocument();
    });
});
