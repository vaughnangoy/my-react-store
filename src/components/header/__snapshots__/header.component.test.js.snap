// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart Dropdown should render 1`] = `
<DocumentFragment>
  <div
    class="header"
  >
    <a
      href="/"
    >
      <svg
        class="logo"
      >
        crown.svg
      </svg>
    </a>
    <div
      class="options"
    >
      <a
        class="option"
        href="/shop"
      >
        SHOP
      </a>
      <a
        class="option"
        href="/shop"
      >
        CONTACT
      </a>
      <a
        class="option"
        href="/sign-in"
      >
        LOGIN
      </a>
      <div
        class="cart-icon"
      >
        <svg
          class="shopping-icon"
        >
          shopping-bag.svg
        </svg>
        <span
          class="item-count"
          data-testid="counter"
        >
          0
        </span>
      </div>
    </div>
    <div
      class="cart-dropdown"
    >
      <div
        class="cart-items"
      />
      <button
        class=" custom-button"
      >
        GO TO CHECKOUT
      </button>
    </div>
  </div>
</DocumentFragment>
`;
